src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberMethodSymbol.cs

```diff
@@ -93,9 +111,25 @@ public MethodKind MethodKind
                 get { return (MethodKind)((_flags >> MethodKindOffset) & MethodKindMask); }
             }
 
+            public RefKind RefKind
+            {
+                get { return (RefKind)((_flags >> RefKindOffset) & RefKindMask); }
+            }
+
+            public bool HasAnyBody
+            {
+                get { return (_flags & HasAnyBodyBit) != 0; }
+            }
+
+            public bool IsExpressionBodied
+            {
+                get { return (_flags & IsExpressionBodiedBit) != 0; }
+            }
+
             public bool IsExtensionMethod
             {
                 get { return (_flags & IsExtensionMethodBit) != 0; }
+                set { ThreadSafeFlagOperations.Set(ref _flags, value ? IsExtensionMethodBit : 0); }
```

Comment:
>set { ThreadSafeFlagOperations.Set(ref _flags, value ? IsExtensionMethodBit : 0); } [](http://example.com/codeflow?start=16&length=83)

Why is this needed? If the setter is used, it has the same issues as the setter we used to have for ```IsVarArg```.

Reply:
Removed.